### PR TITLE
Add unique constraints for Models lacking them

### PR DESCRIPTION
**Pull request by aperrin66**
Solves #81
I also:
  - did some cleanup in the style of the `vocabularies.tests` module, and added doctrings.
  - added the possibility to run a single test module in `runtest.py`

### CODE REVIEW

File: geospaas/vocabularies/tests.py [hunk 1]
```diff
@@ -234,6 +445,8 @@ def setUp(self):
             for model in models]
 
     def test_command_update_vocabularies(self):
+        """Check that the command does not update the vocabularies if they are present"""
```
Comment (aperrin66) [inline geospaas/vocabularies/tests.py]: @akorosov is this actually correct? The code is pretty convoluted.

File: geospaas/vocabularies/tests.py [hunk 2]
```diff
@@ -243,10 +456,15 @@ def test_command_update_vocabularies(self):
             mock.assert_not_called()
 
     def test_command_update_vocabularies_force(self):
+        """
+        Check that the command updates the vocabularies even if they are present when --force is
+        specified
+        """
+
```
Comment (aperrin66) [inline geospaas/vocabularies/tests.py]: @akorosov Same here, is this correct?

File: geospaas/catalog/tests.py [hunk 3]
```diff
@@ -185,6 +186,17 @@ def test_geographiclocation(self):
             geometry=Polygon(((0, 0), (0, 10), (10, 10), (0, 10), (0, 0))))
         geolocation.save()
 
+    def test_unique_constraint(self):
```
Comment (akorosov) [inline geospaas/catalog/tests.py:189]:
```suggestion
    def test_unique_constrain_geographic_location(self):
```

File: geospaas/vocabularies/models.py [hunk 4]
```diff
@@ -29,6 +29,13 @@ def __str__(self):
 
 
 
+    class Meta:
```
Comment (akorosov) [inline geospaas/vocabularies/models.py:32]: Can you remove two empty lines before Meta here and below?

File: geospaas/catalog/models.py [hunk 5]
```diff
@@ -33,6 +33,11 @@ class GeographicLocation(geomodels.Model):
     def __str__(self):
         return str(self.geometry.geom_type)
 
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(name='unique_geographic_location', fields=['geometry'])
```
Comment (akorosov) [inline geospaas/catalog/models.py:38]: So it works this way both in spatialite and PostGIS?

File: geospaas/vocabularies/tests.py (hunk 2 above)
Comment (akorosov) [reply geospaas/vocabularies/tests.py:463]: Yes, here it is correct.

File: geospaas/vocabularies/tests.py (hunk 1 above)
Comment (akorosov) [reply geospaas/vocabularies/tests.py:448]: Yes, it is correct.

**Comment by akorosov** [comment]
That's incredibly good work! Extra thanks for cleaning up tests, @aperrin66 !

So the UniqueContstrain on geometry works both for spatialite and PotsGIS?

If you can just remove two empty line before some Meta() definitions in vocabularies - that would be great. Otherwise I gladly approve :)

File: geospaas/catalog/models.py (hunk 5 above)
Comment (aperrin66) [reply geospaas/catalog/models.py:38]: Yes, but I had to do some tinkering with the migration to make it work: I manually add a unique index when the database is sqlite (according to sqlite [documentation](https://www.sqlite.org/lang_createtable.html), it is equivalent to adding a unique constraint).

File: geospaas/catalog/tests.py (hunk 3 above)
Comment (aperrin66) [reply geospaas/catalog/tests.py]: done

File: geospaas/vocabularies/models.py (hunk 4 above)
Comment (aperrin66) [reply geospaas/vocabularies/models.py:30]: done

**Comment by akorosov** [comment]
@aperrin66 , I approved it. So you can go ahead and merge into master whenever you feel like it.

**Pull request merged**